src/main/java/com/iri/multithreading/Main.java

```diff
@@ -0,0 +1,15 @@
+package com.iri.multithreading;
+
+import com.iri.multithreading.service.Counter;
+import com.iri.multithreading.service.ExtendingThread;
+import com.iri.multithreading.service.ImplementingRunnable;
+
+public class Main {
+    public static void main(String[] args) {
+        Counter counter = new Counter();
+        ExtendingThread extendingThread = new ExtendingThread(counter);
```

Comment:
```suggestion
        Thread extendingThread = new ExtendingThread(counter);
```
remember to use abstract types if it's possible

Reply:
done